kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/metrics/TransactionReportSuite.scala

```diff
@@ -0,0 +1,378 @@
+   * @param generateCommitActions given a [[Transaction]] and engine generates actions to commit
+   * @param expectException whether we expect committing to throw an exception, which if so, is
+   *                        caught and returned with the other results
+   * @return (TransactionReport, durationToCommit, SnapshotReportIfPresent, ExceptionIfThrown)
+   */
+  def getTransactionAndSnapshotReport(
+    getTransaction: Engine => Transaction,
+    generateCommitActions: (Transaction, Engine) => CloseableIterable[Row],
+    expectException: Boolean
+  ): (TransactionReport, Long, Option[SnapshotReport], Option[Exception]) = {
+    val timer = new Timer()
+
+    val (metricsReports, exception) = collectMetricsReports(
+      engine => {
+        val transaction = getTransaction(engine)
+        val actionsToCommit = generateCommitActions(transaction, engine)
+        timer.time(() => transaction.commit(engine, actionsToCommit)) // Time the actual operation
+      },
+      expectException
+    )
+
+    val transactionReports = metricsReports.filter(_.isInstanceOf[TransactionReport])
+    assert(transactionReports.length == 1, "Expected exactly 1 TransactionReport")
+    val snapshotReports = metricsReports.filter(_.isInstanceOf[SnapshotReport])
+    assert(snapshotReports.length <= 1, "Expected at most 1 SnapshotReport")
+    (transactionReports.head.asInstanceOf[TransactionReport], timer.totalDurationNs(),
+      snapshotReports.headOption.map(_.asInstanceOf[SnapshotReport]), exception)
+  }
+
+  /**
+   * Builds a transaction using `buildTransaction` for the table at the provided path. Commits
+   * to the transaction the actions generated by `generateCommitActions` and collects any emitted
+   * [[TransactionReport]]. Checks that the report is as expected
+   *
+   * @param generateCommitActions function to generate commit actions from a transaction and engine
+   * @param path table path to commit to
+   * @param expectException whether we expect committing to throw an exception
+   * @param expectedSnapshotVersion expected snapshot version for the transaction
+   * @param expectedNumAddFiles expected number of add files recorded in the metrics
+   * @param expectedNumRemoveFiles expected number of remove files recorded in the metrics
+   * @param expectedNumTotalActions expected number of total actions recorded in the metrics
+   * @param expectedCommitVersion expected commit version if not `expectException`
+   * @param expectedNumAttempts expected number of commit attempts
+   * @param buildTransaction function to build a transaction from a transaction builder
+   * @param engineInfo engine info to create the transaction with
+   * @param operation operation to create the transaction with
+   */
+  // scalastyle:off
+  def checkTransactionReport(
+    generateCommitActions: (Transaction, Engine) => CloseableIterable[Row],
+    path: String,
+    expectException: Boolean,
+    expectedSnapshotVersion: Long,
```

Comment:
`expectedBaseSnapshotVersion` -- makes it clearer that this is the base/start version, not the end version after the transaction completes. also this aligns better with `transactionReport.getBaseSnapshotVersion`